SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/AppCenter.cs

```diff
@@ -405,7 +401,18 @@ internal void StartInstance(params Type[] services)
             // Enqueue a log indicating which services have been initialized
             if (serviceNames.Count > 0)
             {
-                Task.Run(async () => await SendStartServiceLog(serviceNames));
+                if (InstanceEnabled)
```

Comment:
not sure to follow the fix for potential race conditions by reading github diffs.

Is every code inside the lock statements?

Reply:
reading entire file it looks like it's okay now, will test again on puppet